.vscode/settings.json

```diff
@@ -0,0 +1,5 @@
+// Place your settings in this file to overwrite default and user settings.
+{
+  "editor.tabSize": 4,
+  "editor.insertSpaces": true
```

Comment:
This is unrelated but I noticed half of the files are using 2 spaces and others using 4.  Going to reformat everything to 2 space after the code review is complete, but didn't want to do it now since it introduces lot of white-space change noise in the review.